telegram/chatinvitelink.py

```diff
@@ -0,0 +1,101 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser Public License for more details.
+#
+# You should have received a copy of the GNU Lesser Public License
+# along with this program.  If not, see [http://www.gnu.org/licenses/].
+"""This module contains an object that represents an invite link for a chat."""
+import datetime
+from typing import TYPE_CHECKING, Any, Optional
+
+from telegram import TelegramObject, User
+from telegram.utils.helpers import from_timestamp, to_timestamp
+from telegram.utils.types import JSONDict
+
+if TYPE_CHECKING:
+    from telegram import Bot
+
+
+class ChatInviteLink(TelegramObject):
+    """This object represents an invite link for a chat.
+
+    Objects of this class are comparable in terms of equality. Two objects of this class are
+    considered equal, if their :attr:`invite_link` and :attr:`status` are equal.
+
+    Args:
+        invite_link (:obj:`str`): The invite link.
+        creator (:class:`telegram.User`): Creator of the link.
+        is_primary (:obj:`bool`): True, if the link is primary.
+        is_revoked (:obj:`bool`): True, if the link is revoked.
+        expire_date (:class:`datetime.datetime`, optional): Date when the link will expire or
+            has been expired.
+        member_limit (:obj:`int`, optional): Maximum number of users that can be members of the
+            chat simultaneously after joining the chat via this invite link; 1-99999.
+
+    Attributes:
+        invite_link (:obj:`str`): The invite link. If the link was created by another chat
+            administrator, then the second part of the link will be replaced with ``'…'``.
+        creator (:class:`telegram.User`): Creator of the link.
+        is_primary (:obj:`bool`): True, if the link is primary.
+        is_revoked (:obj:`bool`): True, if the link is revoked.
+        expire_date (:class:`datetime.datetime`): Optional. Date when the link will expire or
+            has been expired.
+        member_limit (:obj:`int`): Optional. Maximum number of users that can be members
+            of the chat simultaneously after joining the chat via this invite link; 1-99999.
+
+    .. versionadded:: 13.4
+
+    """
+
+    def __init__(
+        self,
+        invite_link: str,
+        creator: User,
+        is_primary: bool,
+        is_revoked: bool,
+        expire_date: datetime.datetime = None,
+        member_limit: int = None,
+        **_kwargs: Any,
+    ):
+        # Required
+        self.invite_link = invite_link
+        self.creator = creator
+        self.is_primary = is_primary
+        self.is_revoked = is_revoked
+
+        # Optionals
+        self.expire_date = expire_date
+        self.member_limit = member_limit
+
+        self._id_attrs = (self.invite_link, self.creator)
```

Comment:
I'd vote to put all the required args in here

Reply:
this shouldn't make a difference, since the same link will always have the same `is_primary` and `is_revoked` attributes.

Or wait, do bots know if a link is revoked from someone else? hmm...